camel-openmrs/src/main/java/org/openmrs/eip/component/camel/OpenmrsAttachmentProducer.java

```diff
@@ -0,0 +1,33 @@
+package org.openmrs.eip.component.camel;
+
+import java.io.File;
+
+import javax.activation.DataHandler;
+import javax.activation.FileDataSource;
+
+import org.apache.camel.Exchange;
+import org.apache.camel.attachment.AttachmentMessage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.ApplicationContext;
+
+public class OpenmrsAttachmentProducer extends AbstractOpenmrsProducer {
```

Comment:
Won't this break existing code that assumes only 2 producer subclass? Do you really have to extend AbstractOpenmrsProducer?

Reply:
You should avoid writing a custom component, endpoint or producer, please use a class implementing Processor and put your logic to the process method, this is the purpose of processors

Reply:
Okay. I'll change it to Processor